src/renderer/entities/transaction/lib/transactionService.ts

```diff
@@ -363,6 +369,76 @@ function verifySignature(payload: Uint8Array, signature: HexString, accountId: A
   }
 }
 
+function formatArg(arg: Codec, chain: Chain): unknown {
```

Comment:
Nit: we might want to split this function so it rather uses a set of formatters, each one formatting one thing at a time.
Iirc we have `pipeline` primitive for that or something. Or it can be just a simple array of formatters as well